monetdbe/_cffi/convert/__init__.py

```diff
@@ -72,20 +73,43 @@ def numpy_monetdb_map(numpy_type: np.dtype):
     raise ProgrammingError("append() only support int and float family types")
 
 
-def extract(rcol: monetdbe_column, r: int, text_factory: Optional[Callable[[str], Any]] = None):
-    """
-    Extracts values from a monetdbe_column.
-
-    The text_factory is optional, and wraps the value with a custom user supplied text function.
-    """
-    type_info = monet_c_type_map[rcol.type]
-    col = ffi.cast(f"monetdbe_column_{type_info.c_string_type} *", rcol)
-    if col.is_null(col.data + r):
-        return None
-    else:
-        if type_info.py_converter:
-            result = type_info.py_converter(col.data[r])
-            if rcol.type == lib.monetdbe_str and text_factory:
-                return text_factory(result)
-            return result
-        return col.data[r]
+from monetdbe._cffi.branch import newer_then_jul2021
+if newer_then_jul2021:
```

Comment:
i really hope we can drop these kinds of constructs soon

Reply:
Also I don't expect any API changes anymore. And in the awkward situation that it would happen, I would solve this in a different way.